### PR TITLE
Add Column.name

### PR DESCRIPTION
**Pull request by MarcoGorelli**
This came out of discussions with scikit-learn

I assume it'd be pretty uncontroversial?

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
sure, I've specified, thanks

**Review by kkraus14 — APPROVED** [review]

File: spec/API_specification/dataframe_api/__init__.py [hunk 1]
```diff
@@ -93,6 +95,13 @@ def dataframe_from_dict(data: Mapping[str, Column[Any]]) -> DataFrame:
     Returns
     -------
     DataFrame
+    
+    Raises
+    ------
+    ValueError
+        If any of the columns already has a name, and the corresponding key
+        in `data` doesn't match.
```
Comment (rgommers) [inline spec/API_specification/dataframe_api/__init__.py:103]: Looks good. Can you add the corresponding section to `dataframe_from_dict`? Then this PR looks good to me.

Comment (MarcoGorelli) [reply spec/API_specification/dataframe_api/__init__.py:103]: not sure what mean, isn't this `dataframe_from_dict`?

Comment (rgommers) [reply spec/API_specification/dataframe_api/__init__.py:103]: eh .... ignore me, my brain is fried it seems. I actually opened an editor and stared at something before writing the above 🤦🏼. Resolving and hitting the green button instead.

**Comment by rgommers** [comment]
Thanks Marco!

**Pull request merged**